tests/test_driver/firebase_storage/reference_e2e.dart

```diff
@@ -127,17 +127,23 @@ void setupReferenceTests() {
     });
 
     group('getDownloadURL', () {
-      test('gets a download url', () async {
-        Reference ref = storage.ref('flutter-tests/ok.txt');
-        await ref.putString('ok');
-
-        String downloadUrl = await ref.getDownloadURL();
-        expect(downloadUrl, isA<String>());
-        expect(downloadUrl, contains('ok.txt'));
-        expect(downloadUrl, contains(storage.app.options.projectId));
-      });
+      test(
+        'gets a download url',
+        () async {
+          Reference ref = storage.ref('flutter-tests/ok.txt');
+          await ref.putString('ok');
+
+          String downloadUrl = await ref.getDownloadURL();
+          expect(downloadUrl, isA<String>());
+          expect(downloadUrl, contains('ok.txt'));
+          expect(downloadUrl, contains(storage.app.options.projectId));
+        },
+        // Fails on emulator since iOS SDK 10. See PR notes:
+        // https://github.com/firebase/flutterfire/pull/9708
+        skip: defaultTargetPlatform == TargetPlatform.iOS ||
```

Comment:
Ah - read the notes, also about the `path` thing. That's a shame! I worked really really hard with Paul and the emulator folks a few months back and we had the storage metadata stuff locked down after a couple rounds of "probe until we understand" + "fix either react-native-firebase or firebase-ios-sdk or firebase-tools". There were a few little corner cases on all sides. May take a bit to tack down the regressions, I can help here as well, as react-native-firebase has a very thorough StorageMetadata test setup after all that

Reply:
one other thing to note is since storage has moved to swift and they now use `fatalError` we can no longer catch things like 'emulator already set' etc, as more things move to swift some of our try/catchers for workarounds are going to be more problematic

Reply:
Please tag me in that @russellwheatley - I was neck-deep in it so recently I can hopefully assist